miner/proposal_builder.go

```diff
@@ -380,9 +399,7 @@ func (pb *ProposalBuilder) initSharedData(ctx context.Context, lid types.LayerID
 		pb.shared.beacon = beacon
 	}
 	if pb.shared.active.set == nil {
-		weight, set, err := generateActiveSet(
-			pb.logger,
-			pb.cdb,
+		weight, set, err := pb.generateActiveSet(
```

Comment:
i would prefer not to touch that function at all. instead you can check here if fallback exists and if so always use fallback.

Reply:
The reason why I added it to "generateActiveSet" is to calculate the weights for the ATXs which is not included in the bootstrap file and might not be ready to be calculated when the active set is applied (if ATXs still need to be fetched).

Instead as soon as the active set is used for the first time the total weight is calculated.

Reply:
what i meant is something like. it doesn't change the timing, it is just keeps fromFallback isolated from original code

```go
if fallback, exists := pb.fallback[epoch]; exists {
    set, weight = fromFallback
} else {
   set. weight = generateActiveSet
}
```